src/Modules/OrchardCore.Commerce/Endpoints/Api/CheckoutEndpoint.cs

```diff
@@ -0,0 +1,46 @@
+#nullable enable
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using OrchardCore.Commerce.Endpoints.Permissions;
+using OrchardCore.Commerce.Payment.Abstractions;
+using OrchardCore.Modules;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Endpoints.Api;
+public static class CheckoutEndpoint
+{
+    public static IEndpointRouteBuilder AddCheckoutAsyncEndpoint(this IEndpointRouteBuilder builder)
+    {
+        builder.MapGet("api/Checkout/{shoppingCartId?}", CheckoutAsync)
+            .AllowAnonymous()
+            .DisableAntiforgery();
+
+        return builder;
+    }
+
+    [Authorize(AuthenticationSchemes = "Api")]
+    private static async Task<IResult> CheckoutAsync(
+        string? shoppingCartId,
+        IAuthorizationService authorizationService,
+        HttpContext httpContext,
+        IPaymentService paymentService
+        )
+    {
+        if (!await authorizationService.AuthorizeAsync(httpContext.User, ApiPermissions.CommerceApi) ||
+            !await authorizationService.AuthorizeAsync(httpContext.User, Payment.Permissions.Checkout))
+        {
+            return httpContext.ChallengeOrForbid("Api");
+        }
+
+        if (await paymentService.CreateCheckoutViewModelAsync(shoppingCartId) is not { } checkoutViewModel ||
+            checkoutViewModel.IsInvalid)
```

Comment:
1. If it's invalid then instead of `NotFound` return `TypedResults.Problem` with the same message used [here](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/36676621498f4286b1ece84f0dfe6127a726f1fb/src/Modules/OrchardCore.Commerce.Payment/Controllers/PaymentController.cs#L77). This is a user-facing error so being explicit about it is appropriate.

2. I agree, if it's null we can return an empty cart like
```csharp
var checkoutViewModel = await paymentService.CreateCheckoutViewModelAsync(shoppingCartId) ??
    new CheckoutViewModel(new OrderPart(), Amount.Unspecified, Amount.Unspecified);
```